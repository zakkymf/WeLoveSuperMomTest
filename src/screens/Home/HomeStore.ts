import XLSX from 'xlsx';
import {Alert, PermissionsAndroid} from 'react-native';
import {DownloadDirectoryPath, writeFile} from 'react-native-fs';

export default class Home {
  toLowerCase(str: string) {
    return str?.toLocaleLowerCase();
  }

  toUpperCase(str: string) {
    return str.toUpperCase();
  }

  parseString(value: string) {
    let result = '';
    for (let i = 0; i < value.length; i++) {
      if (i % 2 === 1) {
        result += value[i].toUpperCase();
      } else {
        result += value[i].toLowerCase();
      }
    }

    return result;
  }

  exportCSV(str: string) {
    let result: string = '';
    for (let i = 0; i < str.length; i++) {
      result = result.concat(str[i]);
      if (i < str.length - 1) {
        result = result.concat(',');
      }
    }

    let data = [
      {
        id: new Date(),
        result,
      },
    ];

    let wb = XLSX.utils.book_new();
    let ws = XLSX.utils.json_to_sheet(data);
    XLSX.utils.book_append_sheet(wb, ws, 'data');
    const exportData = XLSX.write(wb, {type: 'binary', bookType: 'xlsx'});

    writeFile(
      DownloadDirectoryPath + '/WeLoveSuperMom.csv',
      exportData,
      'ascii',
    )
      .then(() => {
        Alert.alert('Success', 'Export data successfully');
      })
      .catch(err => {
        Alert.alert('Error', err?.message);
      });
  }

  async checkPermission(word: string) {
    try {
      await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
      ).then(result => {
        if (result === 'granted') {
          this.exportCSV(word);
        }
      });
    } catch (error) {
      return;
    }
  }
}

export const homeStore = new Home();
