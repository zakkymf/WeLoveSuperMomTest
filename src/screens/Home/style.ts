import {StyleSheet} from 'react-native';
import {Colors} from '../../themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    justifyContent: 'center',
  },
  content: {
    padding: 10,
    elevation: 3,
    borderRadius: 8,
    justifyContent: 'center',
    backgroundColor: 'white',
  },
  input: {
    borderBottomWidth: 0.5,
  },
  title: {
    fontSize: 16,
    fontWeight: '500',
    textAlign: 'center',
    color: Colors.black,
  },
  result: {
    fontSize: 12,
    color: Colors.black,
    marginVertical: 5,
  },
});

export default styles;
