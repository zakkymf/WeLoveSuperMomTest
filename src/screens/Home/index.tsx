import React, {useState} from 'react';
import {View, Text, TextInput} from 'react-native';
import Button from '../../components/Button';
import {homeStore} from './HomeStore';
import styles from './style';
const Home = () => {
  const [word, setWord] = useState('hello world');

  const onChangeText = (value: string) => {
    setWord(value);
  };

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <TextInput
          style={styles.input}
          value={word}
          onChangeText={onChangeText}
          placeholder="Insert Word"
        />
        <Button
          label="Export to CSV"
          onPress={() => homeStore.checkPermission(word)}
        />

        <Text style={styles.title}>Result</Text>
        <Text style={styles.result}>{homeStore.toLowerCase(word)}</Text>
        <Text style={styles.result}>{homeStore.toUpperCase(word)}</Text>
        <Text style={styles.result}>{homeStore.parseString(word)}</Text>
      </View>
    </View>
  );
};

export default Home;
