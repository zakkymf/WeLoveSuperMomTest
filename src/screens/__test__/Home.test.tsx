import renderer from 'react-test-renderer';
import Home from '../Home';

describe('Home Screen', () => {
  it('render homescreen correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
