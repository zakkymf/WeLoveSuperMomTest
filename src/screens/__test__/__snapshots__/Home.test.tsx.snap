// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen render homescreen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "elevation": 3,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Insert Word"
      style={
        Object {
          "borderBottomWidth": 0.5,
        }
      }
      value="hello world"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E8157E",
          "borderRadius": 8,
          "height": 48,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        Export to CSV
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
          "textAlign": "center",
        }
      }
    >
      Result
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "marginVertical": 5,
        }
      }
    >
      hello world
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "marginVertical": 5,
        }
      }
    >
      HELLO WORLD
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "marginVertical": 5,
        }
      }
    >
      hElLo wOrLd
    </Text>
  </View>
</View>
`;
